diff --git a/src/meson.build b/src/meson.build
index 645ff311..2e1a90d1 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -15,12 +15,12 @@ gmp_dep = dependency('gmp',
 # get the static dependency, but will return the shared one if the static is
 # not available.
 if build_static and gmp_dep.found()
-  gmp_dep = cpp_compiler.find_library('gmp',
-                                      has_headers: ['gmp.h', 'gmpxx.h'],
-                                      static: true,
-                                      required: false)
+  static_gmp_dep = cpp_compiler.find_library('gmp',
+                                             has_headers: ['gmp.h', 'gmpxx.h'],
+                                             static: true,
+                                             required: false)
   # If we don't find the static library, build the GMP subproject as fallback.
-  if not gmp_dep.found()
+  if not static_gmp_dep.found()
     gmp_dep = subproject('gmp-6.3.0').get_variable('gmp_dep')
   endif
 endif
